src/pages/git/git-pull/index.md

```diff
@@ -3,13 +3,74 @@ title: Git Pull
+  - Check this by executing `git remote -v`
+  - If there are multiple remotes, `git pull` might not be enough information. You might need to enter `git pull origin` or `git pull upstream`. 
+- The branch you are currently checked out to has a corresponding remote tracking branch
+  - Check this by executing `git status`. If there is no remote tracking branch, Git doesn't know where to pull information _from_. 
+  
+  
+### Distributed Version Control
+Git is a **Distributed Version Control System** (DVCS). With DVCS, developers can be working on the same file at the same time in separate environments. After _pushing_ code up to the shared remote repository, other developers can _pull_ changed code.
+
+#### Network interactions in Git 
+There are only four commands that prompt network interactions in Git. A local repository has no awareness of changes made on the remote repository until there is a request for information. And, a remote repository has no awareness of local changes until commits are pushed.
+
+The four network commands are:
+- `git clone`
+- `git fetch`
+- `git pull`
+- `git push`
+
+#### Branches in DVCS 
+When working with Git, it can feel like there are lots of copies of the same code floating all over the place. There are different versions of the same file on each branch. And, different copies of the same branches on every developer's computer and on the remote. To keep track of this, Git uses something called **remote tracking branches**. 
+
+If you execute `git branch --all` within a Git repository, remote tracking branches appear in red. These are read only copies of the code as it appears on the remote. ( When was the last network interaction that would have brought information locally? Remember when this information was last updated. The information in the remote tracking branches reflects the information from that interaction.) 
+
+With **remote tracking branches**, you can work in Git on several branches without network interaction.  Every time you execute `git pull` or `git fetch` commands, you update **remote tracking branches**.
+
+### git fetch plus git merge
+
+`git pull` is a combination command, equal to `git fetch` + `git merge`. 
+
+#### git fetch
+On its own, `git fetch` updates all the remote tracking branches in local repository. No changes are actually reflected on any of the local working branches. 
+
+#### git merge
+Without any arguments, `git merge` will merge the corresponding remote tracking branch to the local working branch. 
+
+#### git pull
+`git fetch` updates remote tracking branches. `git merge` updates the current branch with the corresponding remote tracking branch. Using `git pull`, you get both parts of these updates. But, this means that if you are checked out to `feature` branch and you execute `git pull`, when you checkout to `master`, any new updates will not be included. Whenever you checkout to another branch that may have new changes, it's always a good idea to execute `git pull`. 
```

Comment:
> But, this means that if you are checked out 

Just remote the but, so, This means ....

Reply:
> any new updates will not be included.

Should we mention that git will let them know they are behind? For instance, if you do the `git pull` from `feature` when you checkout to `master` it will let you know if you are ahead / behind your remote.